### ISSUE TITLE
maliit-keyboard not opening on debian testing

### ISSUE THREAD

**Issue by Revilum** [maliit/keyboard#104]
Hi,
after installing maliit-keyboard (and maliit-framework) on debian (compile went without any errors) the keyboard refuses to show up.
Fiddling with the virtual keyboard KDE settings didn't change anything.
I attached the console log when I run `maliit-keyboard`
Any help would be greatly appreciated!

[maliit-keyboard.log](https://github.com/maliit/keyboard/files/8104029/maliit-keyboard.log)

**Comment by dobey** [comment]
Is this under a wayland session? or X11? Is maliit installed into the same prefix as Qt? Under X11, you will need `QT_IM_MODULE=maliit` set in the environment for Qt apps to work with maliit and have the keyboard show up when focusing a text field.

The keyboard itself does not show up always while it is running.

**Comment by Revilum** [comment]
I'm using wayland.
How do I set an environment var for Qt apps? when I start with `env QT_IM_MODULE=maliit dolphin` the keyboard still doesn't show up. I also ran  ```
export QT_IM_MODULE=Maliit
export GTK_IM_MODULE=Maliit
``` as instructed in the maliit-framework README
How do I know which prefix my Qt installed to? I just ran `cmake -DCMAKE_INSTALL_PREFIX=/usr ..`
for both the maliit-framework and maliit-keyboard.

**Comment by dobey** [comment]
On Wayland, the environment variable doesn't need to be set. If you're using Qt from the debian packages, it is installed in `/usr` so you should be fine. How is `maliit-keyboard` being run? Is it being started by `kwin_wayland` or are you trying to run it manually?

**Comment by Revilum** [comment]
There is an icon in the system-tray to enable/disable the onscreen keyboard. When I enable it, I can see the maliit-keyboard process starting in the background. I have tried both using the icon and just running `maliit-keyboard` and then clicking on  a text field (tried QT GTK and native/Xwayland if that makes a difference)

**Comment by dobey** [comment]
If you have a touch screen, you should be able to tap on the text entry instead, to bring up the keyboard, under Wayland. kwin_wayland recently got a change to only request the OSK if the event focusing the text field was a touch event. I'm not sure if there's a setting to disable that so that it will work with mouse events or not.

**Comment by Revilum** [comment]
I tried clicking on a textfield with a mouse/touchscreen/stylus and nothing worked on wayland. Maybe it's an issue with recognizing  which device I used to make the input? There might be some issues with not having the most recent packages installed as debian is not as up to date as it could be.

**Comment by dobey** [comment]
Oh, hrmm, I think it should just work with what's currently in Debian testing, since it's 5.23.5, and I don't think the touch event stuff was backported to that. I'm not sure what would cause it to not open then.

**Comment by Revilum** [comment]
This might be a debian issue actually. I just tested `qtvirtualkeyboard` and when the the keyboard opens on wayland it instantly closes with the message: `qt.qpa.wayland: qtvirtualkeyboard currently is not supported at client-side, use QT_IM_MODULE=qtvirtualkeyboard at compositor-side.`

this looks like an issue with the qt-wayland packages in debian.
I haven't been able to find any further information on this issue.
qtvirtualkeyboard works on X11 (like maliit) without problems

**Comment by hp27596** [comment]
I'm experiencing this issue as well on the latest kde fedora 36 wayland build. Text entry was working on the live iso, but after a full system install, maliit keyboard stopped popping up. If I start maliit server and use environment variables (QT_IM_MODULE and such), I can get the keyboard to pop up again one some qt/gtk applications, but it's a no go on the lock screen. 

I also get this error `qt.qpa.wayland: qtvirtualkeyboard currently is not supported at client-side, use QT_IM_MODULE=qtvirtualkeyboard at compositor-side.` when I try to open a virtual keyboard open wayland as well, so I suspect it might be related to a kde/wayland update (again since osk was working perfectly on the live kde fedora image)

**Comment by Revilum** [comment]
looks like the keyboard was just added to the official debian repositories. Maybe some other people will have the same issue.
https://packages.debian.org/sid/maliit-keyboard

**Comment by dobey** [comment]
With kwin from git master, you can add `export KWIN_IM_SHOW_ALWAYS=1` in a new `/eetc/profile.d/kwin-im-show.sh` and restart, and then maliit should always appear when an entry is focused, regardless of the event type causing it.

**Comment by Revilum** [comment]
ok thanks
the problem is now resolved because my ssd broke and I had by a new one and then reinstall my system. Now everything works as expected... I don't know what my issue was. Probably because I compiled the code manually and didn't do everything correct.
One last question: Is there a convenient way to edit the keyboard layout? Because the german layout includes extra keys for "äöü" which moves every other key to a slightly different position which is really annoying. Normally these special letters are hidden behind "aou".
Thanks for your help anyways!
